internal/services/controller/crd/types.go

```diff
@@ -0,0 +1,50 @@
+package crd
```

Comment:
package name is very generic, we should name it `recommendation` as it holds recommendation types

Reply:
I kept the generic name crd as we might add more custom types in the future, not sure how realistic is that 😄

Reply:
@varnastadeus if I rename the package to `recommendation` intellisense starts beating me up for types `RecommendationSpec` and `RecommendationStatus` as they start with the package name 🙈. 
Maybe `autoscalingv1` might be more descriptive 🤔

Reply:
`autoscalingv1` is taken by k8s already, keeping it `crd`.